### ISSUE TITLE
Loss function fails to converge

### ISSUE THREAD

**Issue by BinBinBang-Tools** [chaudatascience/diverse_channel_vit#3]
Hi @author,

I encountered an issue where the loss function fails to converge while attempting to reproduce the experimental results on the Jump-CP dataset.The specific issue is shown in Screenshot 01 (attached below).

The command used to run the program was:
python main.py -m -cn jump_cp_cfg data_chunk=jump_cp_8 tag=jumpcp_demo model=dichavit ++model.patch_size=16 ++model.enable_sample=True ++model.new_channel_inits=[zero] ++model.pretrained_model_name=small ++optimizer.params.lr=0.0004 ++model.temperature=0.07 ++train.num_epochs=100 ++train.save_model=last ++train.batch_size=64 ++eval.batch_size=256 ++eval.every_n_epochs=1 ++eval.eval_subset_channels=True ++eval.skip_eval_first_epoch=True ++optimizer.params.weight_decay=0.04 ++scheduler.params.warmup_t=10 hardware=two_gpus_dp ++model.use_channelvit_channels=True ++train.extra_loss_lambda=1 ++model.proxy_loss_lambda=0.001 ++model.ortho_loss_v1_lambda=0.001 ++model.gamma_s=1 ++model.gamma_d=4 ++model.reverse_pos_pairs=True ++model.orthogonal_channel_emb_init=True ++logging.wandb.run_name=J-test ++train.seed=33978 ++model.hcs_sampling=lowest_cosine_prob ++model.hcs_sampling_temp=1000 

I followed the exact command specified in the README, but adjusted the batch size and training epochs due to hardware constraints.

I've attempted adjusting the learning rate to test if the loss would decrease, but observed no improvement. Based on the paper's results, the Jump-CP dataset should achieve excellent performance. May I ask if you've encountered similar convergence issues during your experiments?

I would be truly grateful for any insights you could share about this issue. Thank you in advance for your time and assistance!

<img width="1190" height="187" alt="Image" src="https://github.com/user-attachments/assets/1305a52b-cbbf-4c06-8989-3ab7225b7226" />

**Comment by chaudatascience** [comment]
Hi,

Could you share how long the model was trained for?
Also, could you post the full training log or learning curve?  I'll have a look.

Best,
Chau

**Comment by BinBinBang-Tools** [comment]
Hi Chau, thank you for your response!Here are the specific details of my training setup.

The commands I used during training:
python main.py -m -cn jump_cp_cfg data_chunk=jump_cp_8 tag=jumpcp_demo model=dichavit ++model.patch_size=16 ++model.enable_sample=True ++model.new_channel_inits=[zero] ++model.pretrained_model_name=small ++optimizer.params.lr=0.0004 ++model.temperature=0.07 ++train.num_epochs=10 ++train.save_model=last ++train.batch_size=8 ++eval.batch_size=8 ++eval.every_n_epochs=1 ++eval.eval_subset_channels=True ++eval.skip_eval_first_epoch=True ++optimizer.params.weight_decay=0.04 ++scheduler.params.warmup_t=10 hardware=two_gpus_dp ++model.use_channelvit_channels=True ++train.extra_loss_lambda=1 ++model.proxy_loss_lambda=0.001 ++model.ortho_loss_v1_lambda=0.001 ++model.gamma_s=1 ++model.gamma_d=4 ++model.reverse_pos_pairs=True ++model.orthogonal_channel_emb_init=True ++logging.wandb.run_name=J-test ++train.seed=33978 ++model.hcs_sampling=lowest_cosine_prob ++model.hcs_sampling_temp=1000


I reduced the batch size to 16 and trained for only 10 epochs.
The training status in the final epoch is shown in Figure 01 below，and additional attachments include my training logs.

Thank you again for your response and support!

[debug.log](https://github.com/user-attachments/files/21480275/debug.log)
[debug-internal.log](https://github.com/user-attachments/files/21480274/debug-internal.log)

<img width="2379" height="653" alt="Image" src="https://github.com/user-attachments/assets/02e08c1a-9eee-4f7d-a832-f6d148c3630f" />

**Comment by chaudatascience** [comment]
Hi,

It seems the log files don't contain any metrics related to the training, so I'll use the figures then.

From your figures, the main loss at epoch 2 is around 4.16, and at epoch 10 is 3.74. These numbers look reasonable! 
For a reference,  in my run, the loss at epoch 2 and 10 are 4.17 and 3.73, respectively (figure attached).
I’d suggest letting the training continue, as 10 epochs is still quite early in the process.

<img width="1818" height="621" alt="Image" src="https://github.com/user-attachments/assets/7422287f-8975-431a-b09d-4ac436a436d5" />


Just let me know if you have any updates or questions on this!
Best,
Chau

**Comment by BinBinBang-Tools** [comment]
Hi Chau,
So the issue might be due to insufficient training data, is that correct?I'll configure more training epochs to run the training.

I'm a beginner in this field, and this experiment has been an invaluable learning experience for me. 
Thank you again for your response!

**Comment by chaudatascience** [comment]
Yes, you can just train with more epochs!
From your figures, the loss was still going down, so it should not be an issue there. All you need is more training time.

I'll close this issue thread here, but feel free to make a new one if you have any other questions!

Best,
Chau
